PKHeX/Legality/Checks.cs

```diff
@@ -1356,6 +1374,72 @@ private void verifyAbility()
 
             AddLine(Severity.Valid, V115, CheckIdentifier.Ability);
         }
+        private void verifyAbilityPreCapsule(int[] abilities, int abilval)
+        {
+            if (pkm.Format == 3 && pkm.Version == (int)GameVersion.CXD)
+            {
+                // TODO: Check if origin is Colloseum or XD when GameCube analysis is implemented. 
+                // Only Shadow Colloseum  pokemon could have any of the two gen 3 abilities without matching PID.
+                AddLine(Severity.Valid, V115, CheckIdentifier.Ability);
+                return;
+            }
+
+            var abilities_count = abilities.Where(a => a != 0).Distinct().Count();
+            var AbilityMatchPID = abilities_count == 2;
+            if (pkm.Format >= 4 && pkm.InhabitedGeneration(3) && pkm.Species <= Legal.MaxSpeciesID_3)
+            {
+                // gen3Species will be zero for pokemon with illegal gen 3 encounters, like Infernape with gen 3 "origin"
+                // Do not check for gen 3 pokemon that has evolved into gen 4 species, 
+                // those have evolved in generation 4 or 5 and ability must match PID, not need to check gen 3 data
+                var gen3Species = EvoChainsAllGens[3].LastOrDefault()?.Species ?? 0;
+                if(gen3Species > 0)
+                    AbilityMatchPID = verifyAbilityGen3Transfer(abilities, abilval, gen3Species, abilities_count);
+            }
+            
+            // Gen 4,5 pokemon or gen 3 pokemon evolved in gen 4,5 games, ability must match PID
+            if (AbilityMatchPID && abilities_count == 2 && pkm.AbilityNumber != 1 << abilval)
```

Comment:
`abilities_count == 2` is redundant since AbilityMatchPID (which is the same thing) is already true.